### PR TITLE
refactor(core): Extract client mismatch logic to a standalone module

### PR DESCRIPTION
**Pull request by atomrc**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Refactors the logic for handling mismatch and triggering conversation degradation or legal hold detection.

### CODE REVIEW

File: src/script/conversation/MessageRepository.ts [hunk 1]
```diff
@@ -767,25 +738,6 @@ export class MessageRepository {
       this.updateMessageAsSent(conversation, genericMessage.messageId, syncTimestamp ? sentTime : undefined);
     };
 
-    /**
-     * Once a message has been sent, check the state of the members of this conversation.
-     * If, in this conversation, there are still users that have 0 devices, that could mean they have been silently removed from the team
-     * We need to ask backend if those users have been deleted and, if so, trigger the teamMemberLeave event
-     */
-    const checkMissingTeamMembers = async (): Promise<void> => {
```
Comment (atomrc) [inline src/script/conversation/MessageRepository.ts:787]: This is now handled by the mismatch handler

File: src/script/conversation/MessageRepository.ts [hunk 2]
```diff
@@ -1468,80 +1414,14 @@ export class MessageRepository {
...
-        } else {
-          await Promise.all(
-            Object.entries(usersMap).map(([userId, clients]) =>
-              Promise.all(
-                clients.map(client => this.userRepository.addClientToUser({domain: '', id: userId}, client, false)),
-              ),
-            ),
-          );
-        }
-      }
```
Comment (atomrc) [inline src/script/conversation/MessageRepository.ts:1544]: This all logic is also handled by the mismatch handler

**Review by vovchenso — APPROVED** [review]

File: src/script/conversation/ConversationRepository.ts [hunk 3]
```diff
@@ -179,85 +173,66 @@ export class ConversationRepository {
...
+          usersWithoutClients
+            .filter(user => user.deleted)
+            .map(user =>
+              this.teamMemberLeave(
+                this.teamState.team().id,
+                {
+                  domain: this.teamState.teamDomain(),
+                  id: user.id,
+                },
+                new Date(mismatch.time).getTime() - 1,
```
Comment (vovchenso) [inline src/script/conversation/ConversationRepository.ts:211]: Just for curiosity.. why we use `.getTime() - 1` in some places in this file?

Comment (atomrc) [reply src/script/conversation/ConversationRepository.ts:211]: This is in order to inject the message before the message being sent. 
This is a not so nice hack to have message ordered the right way. 

In a not so distant future, I'd like to have a message buffer that takes care of putting the right timestamp to system messages, but for now we are stuck with this system

Comment (vovchenso) [reply src/script/conversation/ConversationRepository.ts:211]: got it thanks for explanation

**Pull request merged**